### PR TITLE
RELATED: SD-423 Storybook for computeRatio and stackMeasuresToPercent behavior

### PR DESCRIPTION
**Pull request by nadtan**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4150/ : SUCCESS in 5m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5269/ : SUCCESS in 15s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2247/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1604/ : UNSTABLE in 10m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5270/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4151/ : FAILURE in 2m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5271/ : SUCCESS in 9s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2248/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1605/ : UNSTABLE in 10m 48s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5272/ : SUCCESS in 7s

**Comment by nadtan** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4152/ : SUCCESS in 6m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5273/ : SUCCESS in 9s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2249/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1606/ : SUCCESS in 10m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5274/ : SUCCESS in 7s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1059.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1607/ : FAILURE in 34m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5275/ : SUCCESS in 10s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1059.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1608/ : SUCCESS in 16m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5276/ : SUCCESS in 6s

**Comment by nadtan** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1059.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1609/ : ABORTED in 1m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5278/ : SUCCESS in 11s

File: stories/internal/ComputeRatio.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const MEASURE_1_RATIO = () =>
+    measure("/gdc/md/storybook/obj/1")
+        .localIdentifier("m1")
+        .ratio();
+
+const MEASURE_2 = () => measure("/gdc/md/storybook/obj/2").localIdentifier("m2");
+
+const MEASURE_2_RATIO = () => MEASURE_2().ratio();
+
+const MEASURE_3 = () => measure("/gdc/md/storybook/obj/3").localIdentifier("m3");
```
Comment (lanhq) [inline stories/internal/ComputeRatio.tsx]: Measures should be created here and reuse it. We don't need to create in each tests.

Comment (nadtan) [reply stories/internal/ComputeRatio.tsx]: Unfortunately, the input measure might be changed in each test by function `disableBucketItemComputeRatio` so when reuse it on other tests, the visualization component can receive wrong measure

Comment (lanhq) [reply stories/internal/ComputeRatio.tsx]: so keep it as is

**Pull request closed without merging**